### ISSUE TITLE
Timeout to short in private VPC for DNS challenge

### ISSUE THREAD

**Issue by serofax** [xenolf/lego#498]
Lego does not work in private AWS VPCs.
The timeout in line https://github.com/xenolf/lego/blob/bacb545c7a26a247cf36a0c9ec46b2697a65465a/acme/dns_challenge.go#L107 needs to be configurable. 

```console
./lego --dns-timeout=3 --server=https://acme-staging.api.letsencrypt.org/directory -d *** --dns=route53 -m test@example.org run
2018/03/12 11:03:52 [INFO][***] acme: Obtaining bundled SAN certificate
2018/03/12 11:03:52 [INFO][***] AuthURL: https://acme-staging.api.letsencrypt.org/acme/authz/***
2018/03/12 11:03:52 [INFO][***] acme: Trying to solve DNS-01
2018/03/12 11:05:41 [INFO][***] Checking DNS record propagation using [10.130.156.130:53]
2018/03/12 11:08:13 [***] Could not obtain certificates
        Time limit exceeded. Last error: read udp 10.130.156.171:44781->205.251.195.58:53: i/o timeout
```
10.130.156.130 is the aws DNS server which is responsible for our VPC
205.251.195.58 is one of the 4 NS of route53

Our machine 10.130.156.171 only has access via a http proxy to the internet. DNS queries DNS server in the internet like (8.8.8.8) or our NSs like 205.251.195.58 are not possible. We can only ask our VPC DNS server (10.130.156.130).

I did some tests in our environment. It takes approx. 5 minutes till the TXT entry is answered first by 205.251.195.58 (not accessible in the VPC) and then max 1 minute later by 10.130.156.130 (accessible in the VPC). With very short refresh rates in our SOA entry we can get it down to approx. 3 minutes till the entry is answered by our dns server. This is still too long.

**Comment by xenolf** [comment]
Hey!

There is currently no way for the user himself to override this timeout but for the provider implementation by having the provider be a `ChallengeProviderTimeout` instead of just a `ChallengeProvider`. You are using route53 and it seems like there was no custom timeout specified for this provider.
What timeout would you feel is plenty for a route53 VPC setup? Like 15 to 30 minutes?

**Comment by serofax** [comment]
Hi,

It depends which refresh rates and cache timings are configured on the SOA entries. But generally 10 minutes should be enough for the last check before triggering the challenge by letsencrypt. So setting the default timeout to 10 minutes is the simplest way.

Increasing should be no problem because this is only for private VPC. VPC with internet access should have no disadvantages.

**Comment by ldez** [comment]
Closed by #603 and #634
